app/src/main/java/com/plenart/organizeme/adapters/BoardItemsAdapter.kt

```diff
@@ -12,45 +12,57 @@ import com.plenart.organizeme.models.Board
 
 open class BoardItemsAdapter(private val context: Context, private val list: ArrayList<Board>): RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
-    private var boardItemClickListener: BoardItemClickInterface? = null;
-
+    private var boardItemClickListener: BoardItemClickInterface? = null
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
-        val binding = ItemBoardBinding.inflate(LayoutInflater.from(parent.context),parent,false);
+        val binding = ItemBoardBinding.inflate(LayoutInflater.from(parent.context),parent,false)
         return BoardItemViewHolder(binding);
     }
 
     override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
-        val model = list[position];
+        val model = list[position]
 
         if(holder is BoardItemViewHolder){
-            Glide.with(context)
-                .load(model.image)
-                .centerCrop()
-                .placeholder(R.drawable.ic_board_place_holder)
-                .into(holder.binding.ivBoardImageItemBoard);
-
-            holder.binding.tvNameItemBoard.text = model.name;
-            holder.binding.tvCreatedByItemBoard.text = "Created by: ${model.createdBy}";
+            holder.bindImage(model.image)
+            holder.bindText(model.name, model.createdBy)
+            holder.bindListeners(position,model)
 
-            holder.itemView.setOnClickListener {
-                if(boardItemClickListener != null){
-                    boardItemClickListener!!.onClick(position, model);
-                }
-            }
         }
     }
 
     override fun getItemCount(): Int {
-        return list.size;
-
+        return list.size
     }
 
     fun setOnClickListener(onClickInterface: BoardItemClickInterface){
-        this.boardItemClickListener = onClickInterface;
+        this.boardItemClickListener = onClickInterface
     }
 
     inner class BoardItemViewHolder(val binding:ItemBoardBinding):RecyclerView.ViewHolder(binding.root){
+
+        fun bindImage(image: String){
+            Glide.with(context)
+                .load(image)
+                .centerCrop()
+                .placeholder(R.drawable.ic_board_place_holder)
+                .into(binding.ivBoardImageItemBoard)
+        }
+
+        fun bindText(name: String, createdBy: String){
+            binding.apply{
+                tvNameItemBoard.text = name
+                tvCreatedByItemBoard.text = "Created by: $createdBy"
+            }
+        }
+
+        fun bindListeners(position: Int, model:Board){
+            itemView.setOnClickListener{
+                if(boardItemClickListener != null){
+                    boardItemClickListener!!.onClick(position,model)
```

Comment:
avoid `!!` -- boardItemClickListener?.onClick(position,model)

Reply:
Okay. I was using non-null assertion to throw NPE to see what value is null and fix it.